### ISSUE TITLE
Parser error occurs when attempting to access constant or static members of statically typed preloaded constants.

### ISSUE THREAD

**Issue by hungpt-a8e** [godotengine/godot#89077]
### Tested versions

v4.2.1.stable.official [b09f793f5]

### System information

Godot v4.2.1.stable - Arch Linux - Vulkan (Forward+)

### What doesn't work:
Encountering a parser error when accessing constant or static members of statically typed preloaded constants.
### How do you expect it to work instead?
The code should parse and run correctly without encountering any errors.

### Steps to reproduce

These codes work correctly without static typing:
In the file `bar.gd`:
```gdscript
extends Node

const BAZ = "baz"

static var qux = "qux"

var quux = "quux"
```
In the file `foo.gd`:
```gdscript
extends Node

const Bar = preload("res://bar.gd")


func _ready() -> void:
	print(Bar.BAZ)
	print(Bar.qux)
```
When I turn on the Inferred Declaration warning, the editor complains in `foo.gd`:
```gdscript
extends Node

const Bar = preload("res://bar.gd")  # gdscript: (INFERRED_DECLARATION): Constant "Bar" has an implicitly inferred static type.


func _ready() -> void:
	print(Bar.BAZ)
	print(Bar.qux)
```
To address this warning, I add static typing to the code:
```gdscript
extends Node

const Bar: GDScript = preload("res://bar.gd")


func _ready() -> void:
	print(Bar.BAZ)
	print(Bar.qux)
```
However, when I run the project, an error occurs:
`Parser Error: Cannot find member "BAZ" in base "GDScript".`
This error only appears when I attempt to access constant or static members. It works fine with instance members:
```gdscript
extends Node

const Bar: GDScript = preload("res://bar.gd")


func _ready() -> void:
	print(Bar.new().quux)  # code runs without error
```

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/godotengine/godot/files/14468994/mrp.zip)

**Comment by dalexeev** [comment]
> To address this warning, I add static typing to the code:

> ```gdscript
> const Bar: GDScript = preload("res://bar.gd")
> ```

* It removes the metatype and the constant can no longer be used as a type hint. See #88553.

**Comment by akien-mga** [comment]
@dalexeev Is this fully fixed by the combination of #92544 and #92558?

**Comment by dalexeev** [comment]
> Is this fully fixed by the combination of #92544 and #92558?

Yes, I tested this and I think the issue is fully resolved.
